lib/timers.js

```diff
@@ -409,6 +393,26 @@ function processImmediate() {
   }
 }
 
+function tryOnImmediate(immediate, queue) {
+  var threw = true;
+  try {
+    immediate._onImmediate();
+    threw = false;
+  } finally {
+    if (threw) {
```

Comment:
I left this as-is because the same style is used in many other `try-catch/finally` blocks throughout the codebase. I don't have a strong opinion one way or the other, but we should be consistent I think.